Shipbreaker Salvage - Salvage Rework Proposal (#575)
space-wizards/docs: src/en/space-station-14/departments/cargo/proposals/shipbreak-salvage.md

# Shipbreak Salvage Proposal

| Designers | Coders | Implemented | GitHub Links |
|---|---|---|---|
| SlamBamActionman | SlamBamActionman et al. | :x: No | TBD |

## Overview

> Shipbreak Salvage (or BreakSalv) is a new primary gameplay loop for the Salvage Specialist job, focusing on pulling in shuttle wrecks to the station and manually disassembling them for valuable resources before sending the remaining structure into a huge industrial shredder for even more materials. It challenges Salvagers in how to deal with a variety of environments, types of hazards and ways to extract materials in an expedient manner, rewarding thorough work with a high resource yield for the station, tool upgrades and bragging rights. BreakSalv emphasizes EVA work on the outside of (but still on!) the station with a sci-fi industrial aesthetic unique to the department.

Reviewer: ```suggestion
Shipbreak Salvage (or BreakSalv) is a new primary gameplay loop for the Salvage Specialist job, focusing on pulling in shuttle wrecks to the station and manually disassembling them for valuable resources before sending the remaining structure into a vast industrial shredder for even more materials. It challenges Salvagers to deal with a variety of environments, types of hazards, and ways to extract materials efficiently, rewarding thorough work with a high resource yield for the station, tool upgrades, and bragging rights. BreakSalv emphasizes EVA work on the outside of (but still on!) the station with a sci-fi industrial aesthetic unique to the department.
```

## Background

### Previous Salvage iterations

> Throughout the years of SS13 and SS14 there have been multiple types of "Salvage Specialist"-type jobs and mechanics, far too many to list exhaustively here. In SS13 the main ones were Shaft Miner, Lavaland and Bitrunner. For SS14, it's been Magnet Pulls, Space Debris, Expeditions and Vgroid. 

Reviewer: ```suggestion
Throughout the years of SS13 and SS14, there have been multiple types of "Salvage Specialist"-type jobs and mechanics, far too many to list exhaustively here. In SS13, the main ones were Shaft Miner, Lavaland, and Bitrunner. For SS14, it's been Magnet Pulls, Space Debris, Expeditions, and Vgroid. 
```

> At its core, Salvage is meant to be a "resource generator" job. A resource generator describes a role that creates new resources for other roles to take advantage of; Cargo is an obvious one with its bounties and cash (turning into purchased products and some materials), and another is Botany with its produce (turning into reagents and some materials). There is an inherent sense of accomplishment in this kind of gameplay as the resources created are appreciated by other players and the supply can influence the round in notable ways. Salvage in SS14 has primarily focused on lathe materials, with some being soft- or hard-locked behind the job such as silver or uranium. BreakSalv intends to continue with this specialization as it fills a niche that other resource generators do not, providing high volumes of basic materials and rare materials difficult to acquire elsewhere.

Reviewer: ```suggestion
At its core, Salvage is meant to be a "resource generator" job. A resource generator is a role that creates new resources for other roles to use; Cargo is an obvious one, with its bounties and cash (turning into purchased products and some materials), and another is Botany, with its produce (turning into reagents and some materials). There is an inherent sense of accomplishment in this kind of gameplay, as other players appreciate the resources created, and the supply can influence the round in notable ways. Salvage in SS14 has primarily focused on lathe materials, with some being soft- or hard-locked behind the job, such as silver or uranium. BreakSalv intends to continue this specialization because it fills a niche that other resource generators do not, providing high volumes of basic and rare materials that are difficult to acquire elsewhere.
```

"Describes a role" sounds a bit off.

> Something that has been a recurring pattern with both SS13 and SS14 implementations of Salvage is their unavailability. Whether it is Lavaland, Expeditions or Vgroid, there has been a strong focus of having Salvage leave the station in favor of a highly challenging environment. This has given Salvage a bit of an identity as being something hardcore PvE players engage with, but also came with the downside that said players effectively "disappeared" from the round, not caring about station events and being difficult to reach for antagonists - or if one of the Salvagers were antagonists, having an extremely easy system to dispose of their fellow Salvagers. This has been felt potently in SS14 where there were periods of the game where Salvage used shuttle gameplay to effectively leave the station for the entire duration of the round, creating their own medical treatment and lathes, getting antagonist-level loot and using Cargo's purchase mechanic to completely bypass interacting with the rest of the station. 

Reviewer: ```suggestion
A recurring pattern in both SS13 and SS14 implementations of Salvage is their unavailability. Whether it is Lavaland, Expeditions, or Vgroid, there has been a strong focus on having Salvage leave the station for a highly challenging environment. This has given Salvage a bit of an identity as being something hardcore PvE players engage with, but also came with the downside that said players effectively "disappeared" from the round, not caring about station events and being difficult to reach for antagonists, or if one of the Salvagers were antagonists, having a straightforward system to dispose of their fellow Salvagers. This has been felt potently in SS14, where there were periods of the game where Salvage used shuttle gameplay to effectively leave the station for the entire duration of the round, creating their own medical treatment and lathes, getting antagonist-level loot, and using Cargo's purchase mechanic to bypass interacting with the rest of the station completely. 
```

> That being said, there have been elements of Salvage that allow unique gameplay, themes and game mechanics to shine through. PvE-focused combat, variation via mapped magnet pulls, mining, dungeon/world generation and shuttle building are some of the things that Salvage has showcased. Not all of these were necessarily conducive to station-aligned gameplay (dungeon/world generation being part of Expeditions is one such example) but it shows Salvage affords a rather large design space that would otherwise be inaccessible through other jobs.

Reviewer: ```suggestion
That said, Salvage has elements that allow unique gameplay, themes, and mechanics to shine through. PvE-focused combat, variation via mapped magnet pulls, mining, dungeon/world generation, and shuttle building are some of the things that Salvage has showcased. Not all of these were necessarily conducive to station-aligned gameplay (dungeon/world generation being part of Expeditions is one such example). Still, it shows Salvage affords a rather ample design space that would otherwise be inaccessible through other jobs.
```

Because of this, BreakSalv will not attempt to capture all the aspects of previous Salvage iterations, but it does use what has come before as an inspiration and launchpad for its new design. As you read this document you will note mechanical similarities to *some* previous implementations of Salvage (most notably Magnet Pulls). That is a result of those systems being a good fit for the ideas core to BreakSalv, rather than BreakSalv building upon them just because they exist.

> As the name suggests Shipbreaker Salvage is also inspired by the game [Hardspace: Shipbreaker](https://store.steampowered.com/app/1161580/Hardspace_Shipbreaker/). It does not try to recreate the game's mechanics in SS14, but rather attempts to emulate the feeling of engagement and fulfilment that breaking apart a spaceship provides in the same manner as the game does. The game provides a good aesthetic and emotional goalpost for what BreakSalv attempts to be (and it's a pretty good game!).

Reviewer: ```suggestion
As the name suggests, Shipbreaker Salvage is also inspired by the game [Hardspace: Shipbreaker](https://store.steampowered.com/app/1161580/Hardspace_Shipbreaker/). It does not try to recreate the game's mechanics in SS14; instead, it aims to emulate the feeling of engagement and fulfilment that breaking apart a spaceship provides, as the game does. The game provides a good aesthetic and emotional goalpost for what BreakSalv attempts to be (and it's a pretty good game!).
```

## Features to be added

BreakSalv is the primary gameplay loop for the _Salvage Specialist_ job belonging to the Salvage subdepartment of the Cargo department. The goal is to produce resources for distribution to the rest of the station in the form of high volumes of basic materials and rare, unique materials and equipment that are otherwise difficult to procure. This is achieved by spawning in wrecks of old derelict shuttles, extracting rare materials from them by hand and then shredding the remainder of the shuttle for basic materials.

> The work of a Salvager takes place in the _Salvage Bay_, a location exposed to space on the outside of the station in close proximity to the Cargo department. While the majority of work Salvage performs happens inside of the Salvage Bay due to the station-locked machinery there, they should have a small locker room inside of Cargo for suiting up and easy access to the rest of the Cargo department to ease the transfer of materials.

Reviewer: ```suggestion
The work of a Salvager takes place in the _Salvage Bay_, a location exposed to space on the outside of the station, near the Cargo department. While most Salvage work occurs in Salvage Bay due to the station-locked machinery there, they should have a small locker room in Cargo for suiting up, with easy access to the rest of the Cargo department to facilitate material transfers.
```

Redundancy here and text improvement.

BreakSalv has a gameplay loop that can be broken up in three primary parts: retrieval, extraction and shredding. 

### Retrieval

> **Retrieval** is the process of spawning and getting wrecks to the station so that their resources can be extracted.

Reviewer: ```suggestion
**Retrieval** is the process of spawning wrecks and transporting them to the station so their resources can be extracted.
```

> Wrecks are pre-made grids themed around being decommissioned shuttles or destroyed station sections. To get access to wrecks, Salvage has a unique console in the Salvage Bay called the _Salvage Magnet_. 

Reviewer: ```suggestion
Wrecks are pre-made grids themed around being decommissioned shuttles or destroyed station sections. To access wrecks, Salvage has a unique console in Salvage Bay called the _Salvage Magnet_. 
```

#### Salvage Magnet

The Salvage Magnet (or just magnet) provides an interface where Salvagers can select one of a small selection of wrecks that they want to work on called the "Wreck Selection" view. Each entry lists information of what type of wreck it is and what resources are expected to be found on it.

> One of the key pieces of information listed in a magnet entry is the _wreck type_ the wreck is. Types describe some sort of environmental hazard or property of the wreck, providing challenges unique to the wreck type and gives Salvagers variety and choice in the wreck they want work on.

Reviewer: ```suggestion
One of the key pieces of information listed in a magnet entry is the _wreck type_ the wreck is. Types describe environmental hazards or wreck properties that pose challenges unique to each type, giving Salvagers variety and choice in the wreck they want to work on.
```

- **Infested:** Contains a large number of space mobs that will attack the Salvagers. These mobs are resistant to piercing damage, making ballistic weapons ineffective.
- **Overgrown:** Kudzu covers the wreck, along with trees blocking pathways.
- **Radioactive:** Contains a large number of radioactive objects that require quick removal.
- **Defended:** Contains dangerous turrets (ballistic and/or laser) and mines that must be disposed of.
- **Shadowy:** Covered in the visually obscuring shadow anomaly fog that has to be navigated or removed via flashes.
- **Electrified:** Features electrical grilles and shocked airlocks that require either insulated gloves or strategic turning off of APCs.
- **Fortified:** Contains powered access-locked and high-security airlocks with strong walls. These must be hacked or destroyed to reach the resources inside.
- **Anchored:** Has a mini-station anchor inside of it that must be manually turned off before the wreck can be moved.
- **Distant:** Spawns at a further distance away from the station than normal. Jetpack recommended.
- **More!** The idea is that these wreck types can be expanded upon for more variety. 

> Only one wreck may be active with the magnet at any given time. Once a wreck has been selected the interface changes to a new view as described in the [Magnet Evaluation](#magnet-evaluation) section.

Reviewer: ```suggestion
Only one wreck may be active with the magnet at any given time. Once a wreck has been selected, the interface changes to a new view as described in the [Magnet Evaluation](#magnet-evaluation) section.
```

#### Towing

Once a wreck has been spawned, it will be located somewhere in nearby space in close proximity to the Salvage Bay. If the wreck is not within the player's viewing distance, Salvage may use a _Mass Scanner_ mapped in the Bay to locate it. When compared to previous iterations of Salvage, the wrecks are meant to spawn much closer to the station. Regardless, Salvage will want to move the wreck closer to the Salvage Bay as it will make it faster to extract materials from it and is a necessity to complete later steps. This is known as _towing_.

> To tow the wreck, Salvage will have access to _grappling hooks_ and _magboots_. While grappling hooks on their own simply drag the user towards the hook, when combined with the grounding force of the magboots Salvagers are able to pull grids towards each other (or in the case one has a station anchor, only the one without gets pulled). Ideally Salvage will be able to stand on the outer rim of the station and use their grappling hooks to drag the wreck until it is within working distance; once it is, they can proceed with the next step.

Author: I'm taking inspiration from the fact that you can already kinda sorta do this with magboots, but with fire extinguishers. If you stand on a grid with magboots and spray, you affect the grid. Can make your own water-powered manual shuttle this way!

I included a stationary "spear gun" in the Further Embodiments section that could be a machine used for this purpose, but I think since salvagers already have use for both magboots and grappling hooks for other parts of their work and that it's fairly common game language to have a grappling hook pull things towards you as much as it does you to them, that it could work for this purpose too. 

You get to shoot your grappling hook at stuff and reel it in. It's awesome!

Wreck resources can not directly be used as materials but need to be processed first. The _recycler_ is a good option for this as it is a station-mounted machine and Salvage can easily have their own. As a rule, wreck resources should _not_ focus on steel or glass materials, since those are primarily obtained through the shredding step.  

- **Two-handed:** Requires two hands to pick up. 
- **Large:** Big enough to require dragging and can't be held or put in containers.
- **Fragile:** Easily broken if thrown around recklessly.
- **Mounted:** Attached to a wall or integrated into some machinery that requires partial deconstruction.
- **Radioactive:** Emits radiation.
- **Electrically Unstable:** If destroyed or processed incorrectly, will emit an EMP pulse.
- **Explosive:** If destroyed or processed incorrectly, will explode.
- **Pulsating:** Occasionally emits a gravity pulse that pushes away other objects around it.
- **More!** The idea is that these wreck resource types can be expanded upon for more variety. 

### Shredding

**Shredding** is the process of finalizing a salvage operation by grinding up the remains of a wreck into base materials.

> Once all valuable wreck resources have been extracted and the Salvagers feel finished with the wreck, it is time to get rid of the wreck itself. For this, the Salvage Bay has a large wall of _industrial shredders_, and _separation charges_.

Reviewer: ```suggestion
Once all valuable wreck resources have been extracted and the Salvagers are finished with the wreck, it is time to dispose of the wreck itself. For this, the Salvage Bay has a large wall of _industrial shredders_, and _separation charges_.
```

#### Industrial Shredders

Industrial shredders work by grinding up any tile that comes in contact with the open front of the machine, destroying it and any structure attached. It is essentially a recycler specializing in grids. As wrecks are mostly made up of metal and glass parts, this becomes the main method Salvage produces steel and glass sheets. To prevent Salvage from simply sending a fresh wreck straight into the industrial shredders they provide a remarkably lower yield for wreck resources and items that can already be recycled, and some wreck resources may harm the shredders if inserted into them.

> Since some individual grid tiles could feasibly be made with less than one sheet of steel material, those tiles only have a percentage chance to produce a sheet. E.g. lattice, which is made of 1 steel rod (equivalent to 0.5 steel sheet) has at most a 50% to produce a sheet.

Reviewer: ```suggestion
Since some individual grid tiles could feasibly be made with less than one sheet of steel material, those tiles only have a percentage chance to produce a sheet. E.g., a lattice, which is made of 1 steel rod (equivalent to 0.5 steel sheet), has at most a 50% chance of producing a sheet.
```

Unless I am reading this wrong, should the word 'chance' be there?

> Since some individual grid tiles could feasibly be made with less than one sheet of steel material, those tiles only have a percentage chance to produce a sheet. E.g. lattice, which is made of 1 steel rod (equivalent to 0.5 steel sheet) has at most a 50% to produce a sheet.

Author: I am definitely open to just making the shredders outright destroy wreck resources or other such materials, should it turn out players are so lazy they just shred the entire thing for single-digit bars. Overall something that can be looked at and discovered in playtesting.

> Each shredder acts as a conveyor belt that is only able to push items perpendicular to its opening; since each Salvage Bay is meant to have a line of shredders (known as the "shredder wall") this means the material can travel along the line to a pick-up location (either in the Bay or in Cargo proper). Shredders are *only active* when there is an active wreck being worked on via the magnet and turn on/off via device linking to it. In the event that a shredder gets damaged (because of meteors or misconduct) it should be repairable, e.g. via welding. 

Reviewer: ```suggestion
Each shredder acts as a conveyor belt that is only able to push items perpendicular to its opening; since each Salvage Bay is meant to have a line of shredders (known as the "shredder wall"), this means the material can travel along the line to a pick-up location (either in the Bay or in Cargo proper). Shredders are *only active* when a wreck is being worked on via the magnet, and they turn on/off via the device that links to it. In the event that a shredder gets damaged (because of meteors or misconduct) it should be repairable, e.g. via welding. 
```

Shredders do not have the safety mechanism of recyclers which means they deal contact damage, but the damage is low enough that it isn't an efficient method of murder. You wouldn't want to stand or walk through a shredder for a long duration but just touching it for a few seconds won't be lethal.

> Shredders are *only* able to shred wrecks attached to the linked salvage magnet. While realistically there shouldn't be a limit, this is an instance where moderation efforts need to be made as having them able to shred any grid would make major griefing and self-antagonism too easy to perform, and also allow crew to easily weaponize the shredder wall against shuttle-based antagonists (despite how awesome it sounds, we do not want crew to mount shredders to a Cargo shuttle to bulldoze the Xenoborg Mothership). There is the possibility to remove this limitation as an EMAG interaction for the salvage magnet however, though if that is to be implemented it should come with a station-wide announcement to alert crew to the impending danger.

Reviewer 2: Shredders are a pretty big mapping concern. It sounds like they'd take up a lot of space, so why not just have the magnet be able to delete chunks of a grid at a time as part of its functionality? Or just have it break apart the entire grid when it's finished in one fell swoop?

> Shredders are *only* able to shred wrecks attached to the linked salvage magnet. While realistically there shouldn't be a limit, this is an instance where moderation efforts need to be made as having them able to shred any grid would make major griefing and self-antagonism too easy to perform, and also allow crew to easily weaponize the shredder wall against shuttle-based antagonists (despite how awesome it sounds, we do not want crew to mount shredders to a Cargo shuttle to bulldoze the Xenoborg Mothership). There is the possibility to remove this limitation as an EMAG interaction for the salvage magnet however, though if that is to be implemented it should come with a station-wide announcement to alert crew to the impending danger.

Reviewer: ```suggestion
Shredders are *only* able to shred wrecks attached to the linked salvage magnet. While realistically there shouldn't be a limit, this is an instance where moderation efforts need to be made as having them able to shred any grid would make major griefing and self-antagonism too easy to perform, and also allow crew to easily weaponize the shredder wall against shuttle-based antagonists (despite how awesome it sounds, we do not want crew to mount shredders to a Cargo shuttle to bulldoze the Xenoborg Mothership). It is possible to remove this limitation as an EMAG interaction for the salvage magnet; however, if implemented, it should be accompanied by a station-wide announcement to alert the crew to the impending danger.
```

#### Separation Charges

Some wrecks will simply be too large to fit into the Salvage Bay and the shredder wall. Salvage will be tasked to split these wrecks into smaller parts using a new tool called separation charges. These act similar to cables but are placed on the edges of tiles and when activated split the grid into two parts, similar to how an RCD can separate grids. This split is accompanied by a small spark (not an explosion!) and a small push force separating the grids. Large wrecks should be designed with this functionality and in some places have pre-defined ideal spots to place charges.

Separation charges will only activate if the line they create fully separates the two parts of the grid from each other. If the charges do not form a full cut across the grid or there are other sections still connecting the parts together, the charges fail to activate. Since separation charges have potential for some pretty strong antagonistic plays, such as disconnecting vital station sections, placing a single separation charge has a fairly lengthy do-after. 

> It's recommended (but not required) that _weld charges_ are implemented as well, that act opposite to separation charges and give the ability to attach grids together. These would work well as an Engineering tool and/or part of the RCD.

Author: Kinda? Sorta? I encountered a situation just yesterday where a syndie bomb blew up the armory on Fland and for whatever reason a chunk of floor tile was still in the hole. Had to be RCD'd away.

There's also the case against disconnecting AIs, which is mostly redundant these days since most AI attachments have been reinforced to avoid it, but I am concerned that the inclusion of separation charges could maybe bring that back into vogue. 

Regardless the weld charges are just a minor footnote on the impact BreakSalv can have and not really necessary for the gameplay loop to function.

> It's recommended (but not required) that _weld charges_ are implemented as well, that act opposite to separation charges and give the ability to attach grids together. These would work well as an Engineering tool and/or part of the RCD.

Reviewer 3: I had a conversation about MergeGrids (which, while it does exist in the engine, is not used in anything) and we came to a conclusion that fully merging grids is probably too dangerous physics-wise (e.g. if someone makes a walkway between the vgroid and the station it can have wonky effects) -- in #577 i propose a "piton" for rapid docking instead, as well as a system for something like sharing power between grids, which would be safer (and maybe allow for something like towing a grid behind your shittle)

Also, I imagine shipbreakers might want to dock  their grid in place temporarily while working on it

#### Magnet Evaluation

The final part of the BreakSalv loop is Salvage returning to the magnet and getting an evaluation of their work.

While a wreck is being worked on the magnet interface displays a "Magnet Evaluation" view with the following information:

- **Wreck Percentage:** Percentage of how much of the spawned wreck grid has been processed; 0% means all grid tiles remain, 100% means no grid tiles remain.
- **Shredder Efficiency:** Efficiency of connected industrial shredders. This is calculated based on whether the shredders have processed any entities that they give lower yield for, such as wreck resources.
- **Wreck Time:** Time since magnet pull started.

> After some time has passed (short enough to not feel restrictive, but long enough that Salvage can't just cycle wrecks), Salvagers may choose to despawn any remaining grids of their wreck using a "Despawn Wreck" button on the magnet. There is an incentive to fully shred the wreck however. If the Wreck Percentage is at 100% a previously greyed out button "Claim Processing Reward" becomes available. Clicking this button rewards Cargo with a monetary boost and announces in the Supply channel that Salvage finished processing the wreck along with the Shredder Efficiency and Wreck Time. 

Reviewer: ```suggestion
After some time has passed (short enough not to feel restrictive, but long enough that Salvage can't just cycle wrecks), Salvagers may choose to despawn any remaining grids of their wreck using a "Despawn Wreck" button on the magnet. There is, however, an incentive to thoroughly shred the wreck. If the Wreck Percentage is 100%, the previously greyed-out "Claim Processing Reward" button becomes available. Clicking this button rewards Cargo with a monetary boost and announces in the Supply channel that Salvage finished processing the wreck, along with the Shredder Efficiency and Wreck Time. 
```

Once the progress bar has completed the Salvagers are rewarded with a new wreck option in the wreck selection view that features a higher tier of wreck. This higher tier is intended to be more challenging by featuring multiple wreck types at once, more resources and are generally larger.

#### Salvage Specialist equipment

From the start of the round, Salvagers should have access to:

- A salvage hardsuit. Necessary for working in the Salvage Bay.
- Basic tools and toolbelt. For navigating and deconstructing the wreck during the extraction step.
- A survival knife and a portable kinetic accelerator. Used to deal with hostile mobs on wrecks during the extraction step.
- A grappling hook and separation charges. Necessary to complete the retrieval and shredding step.

Upgrades to the Salvager's arsenal could be:

- Improved hardsuit with higher movement speed/defense.
- Welding goggles/gas mask, insulated gloves, improved welding equipment.
- Jaws of life and power drill.
- Simple laser weapons.

## Game Design Rationale

> BreakSalv is intends to position Salvage as a primary resource generator for the station by utilizing the unique game mechanics opportunities afforded by working on the outside of the station. 

Reviewer: ```suggestion
BreakSalv intends to position Salvage as a primary resource generator for the station by leveraging the unique game mechanics opportunities afforded by working outside the station. 
```

### BreakSalv in detail

This section addresses specific design choices with the various features BreakSalv aims to implement.

#### Salvage in space

Previous iterations of Salvage have had a problem with Salvage "fucking off into space" and not engaging with the station for the majority of the round. This has less been the fault of space itself and rather the Salvage mechanics that encouraged leaving the station; shuttle building and space debris encouraged hopping from debris to debris to gather resources, with the vrgoid asteroid being the epitome of encouraging Salvage to leave for extended durations and make "mini-stations" with medical capabilities and even their own lathes replacing the need for the station altogether. Expeditions disconnected Salvage from the station map entirely and fultons meant Salvagers didn't need to return to the station to hand over the materials. 

> BreakSalv locks Salvage to the station grid by making heavy use of the magnet and the industrial shredders. There is little to no reason to engage in gameplay away from the station and by being in so close proximity to the station there isn't a strong incentive to set up station-replacing equipment. This means that Salvage will always be accessible to react to station events and become easier to locate e.g. for antagonists. 

Reviewer: ```suggestion
BreakSalv locks Salvage to the station grid by making heavy use of the magnet and the industrial shredders. There is little to no reason to engage in gameplay away from the station, and by being in so close proximity to the station, there isn't a strong incentive to set up station-replacing equipment. This means that Salvage will always be accessible to react to station events and become easier to locate, e.g., for antagonists. 
```

> There are still some pros and cons with working on the outside of the station. EVA work is a very unique mechanic to the space setting that otherwise only really Engineering deals with as part of their primary gameplay loop, is aesthetically cool and leverages the open space afforded by it. This does limit crew-job interactions however, as there is no Salvage "front desk" and the Salvage Bays would not be by any high-traffic area (unless a station specifically designs its layout around it). On the other hand, tying Salvage's work to the Salvage Bay will make them massively more accessible compared to previous iterations. Paramedics will have an easier time accessing Salvagers in case of lethal explosions, Cargo & QM will generally know where Salvagers are while working, Engineers can pass by Salvagers when traversing the outside of the station and there is a much lower barrier of entry for Passengers to help out by virtue of the Salvage Bay being more open.

Reviewer: ```suggestion
There are still some pros and cons to working outside the station. EVA work is a very unique mechanic to the space setting that otherwise only really Engineering deals with as part of their primary gameplay loop, and it is aesthetically cool and leverages the open space afforded by it. This does limit crew-job interactions, however, as there is no Salvage "front desk" and the Salvage Bays would not be by any high-traffic area (unless a station specifically designs its layout around it). On the other hand, tying Salvage's work to the Salvage Bay will make it massively more accessible compared to previous iterations. Paramedics will have an easier time accessing Salvagers in case of lethal explosions, Cargo & QM will generally know where Salvagers are while working, Engineers can pass by Salvagers when traversing the outside of the station, and there is a much lower barrier of entry for Passengers to help out by virtue of the Salvage Bay being more open.
```

The Salvage Bay also affords a good point for holopads, and with the Wreck Timer being purely cosmetic there is no penalty for Salvagers to pause their work to engage in conversation or do tasks for the station.

#### The Salvage Magnet and Wrecks

Wrecks provide variety to the round, both in the selection of available wrecks but also what contents they may provide. They are also excellent opportunities for environmental story-telling; all wrecks should have some sort of "lore" behind it that ties into its wreck type and what the wreck depicts. Wrecks should not be created through pure random generation but be hand-crafted grids by our talented mappers. Wrecks should ideally not have just a single way that they can play out to create emergent gameplay, and Salvagers should feel they can make meaningful choices with what wrecks they spawn in and also in how they approach them.

> As mentioned in the previous section, the salvage magnet becomes part of what ties Salvagers to the station. Despite its unassuming appearance it is a critical part of ensuring that BreakSalv flows smoothly and a lot of care and consideration needs to be taken to ensure that its presentation and mechanics (such as delays) don't encourage bad or unfun gameplay. 

Reviewer: ```suggestion
As mentioned in the previous section, the salvage magnet becomes part of what ties Salvagers to the station. Despite its unassuming appearance, it is a critical part of ensuring that BreakSalv flows smoothly, and a lot of care and consideration needs to be taken to ensure that its presentation and mechanics (such as delays) don't encourage bad or unfun gameplay. 
```

> A potentially contentious part is the Upgrade Magnet button. Forcing Salvagers to not engage with their primary gameplay loop may be seen as killing the momentum, however considering that it is optional and up to the Salvagers when it should be activated (in case Salvage just wants to do more wrecks right this moment) and that it only unlocks after Salvage has done some wrecks already should hopefully place it at a point where Salvagers are looking for an excuse to take a break. Ideally it will be implemented such that Salvagers can go "We'll do one more wreck and then hit the bar while the magnet upgrades".

Reviewer: ```suggestion
A potentially contentious part is the Upgrade Magnet button. Forcing Salvagers to not engage with their primary gameplay loop may be seen as killing the momentum, however considering that it is optional and up to the Salvagers when it should be activated (in case Salvage just wants to do more wrecks right this moment) and that it only unlocks after Salvage has done some wrecks already should hopefully place it at a point where Salvagers are looking for an excuse to take a break. Ideally, it will be implemented such that Salvagers can go "We'll do one more wreck and then hit the bar while the magnet upgrades".
```

#### Grappling Hooks

> pulling grids is fun idk what else to say yippee pew pew i'm so strong pulling shottle weee

Reviewer 4: <img width="275" height="183" alt="Image" src="https://github.com/user-attachments/assets/ea9b8cfb-499c-476d-9e7e-cc188990f9e6" />

#### Shredders and Wreck resources

> The main motivation behind wreck resources is to encourage Salvage to enter the wreck and loot it properly by hand. By making them focus on containing rare materials they also become something Salvagers will want to seek out (which will be a positive experience), rather than "scrap" for basic materials which the shredding step is already responsible for. 

Reviewer: ```suggestion
The main motivation behind wreck resources is to encourage Salvage to enter the wreck and loot it properly by hand. By making them focus on containing rare materials, they also become something Salvagers will want to seek out (which will be a positive experience), rather than "scrap" for basic materials, which the shredding step is already responsible for. 
```

> By making the wreck resources designed specifically for this we can also avoid Salvagers simply mashing a shuttle straight into the shredder wall, as they can be set to give little to no resources from this. Wreck resources that damage the shredders also strongly discourage Salvagers from sending the wrecks into the shredders, though experienced Salvagers may be able to partially shred a shuttle to open up a section that would be otherwise difficult to get to. 

Reviewer: ```suggestion
By making the wreck resources designed specifically for this, we can also avoid Salvagers simply mashing a shuttle straight into the shredder wall, as they can be set to give little to no resources from this. Wreck resources that damage the shredders also strongly discourage Salvagers from sending the wrecks into the shredders. However, experienced Salvagers may be able to partially shred a shuttle to open a section that would otherwise be difficult to access. 
```

## Roundflow & Player interaction

As previously mentioned, Salvage will have some inherent progression through what is found on the wrecks, but may also be assisted by upgrades from Science in the form of useful equipment. There isn't as much of an endgoal for Salvage as there is an escalation of stakes to match the stronger equipment, with larger volumes of rarer materials and more challenging environments becoming available/feasible as the round progresses.

> Wrecks should ideally refrain from having certain items. This is primarily medical supplies, food and hard cash; Salvagers shouldn't be able to loot wrecks and set up their own medical facilities or sustain themselves completely on wreck food to prevent them becoming independent from the station, and finding large stacks of money invalidates the work Cargo Technicians put in with completing bounties. Salvage should however be able to supply Medical and Service with resources, and may be a core part of completing certain bounties for Cargo and providing items for Cargo Technicians to sell.

Reviewer: ```suggestion
Wrecks should ideally refrain from having certain items. This is primarily medical supplies, food, and hard cash; Salvagers shouldn't be able to loot wrecks and set up their own medical facilities or sustain themselves completely on wreck food to prevent them from becoming independent from the station, and finding large stacks of money invalidates the work Cargo Technicians put in with completing bounties. Salvage should, however, be able to supply Medical and Service with resources, and may be a core part of completing specific bounties for Cargo and providing items for Cargo Technicians to sell.
```

> Salvage has had a tradition of getting "gamer loot" via wrecks, i.e. equipment that would be very strong in the hands of an antagonist (or worse, used to validhunt/shut down antagonists) when they return to the station. Some wrecks, especially those that become available after the magnet upgrade, may contain some stronger equipment and especially cosmetics, but to prevent undesirable gamer loot Syndicate equipment should be heavily restricted from the wrecks.

Reviewer: ```suggestion
Salvage has had a tradition of getting "gamer loot" via wrecks, i.e., equipment that would be very strong in the hands of an antagonist (or worse, used to validhunt/shut down antagonists) when they return to the station. Some wrecks, especially those that become available after the magnet upgrade, may contain some stronger equipment and especially cosmetics, but to prevent undesirable gamer loot, Syndicate equipment should be heavily restricted from the wrecks.
```

In terms of what resources Salvage should be able to uniquely supply to the station, this document primarily cites rare materials. With the content currently available this can negatively impact gameplay for other departments (e.g Science) where Salvage is _required_ to provide materials to enable core gameplay and rewards. The design of BreakSalv is such that new materials and equipment could be introduced as new wreck resources, to make poor/non-existent Salvage work less blocking for other roles. As this would involve a larger rework of the overall resource system in the game with content supporting it, and that it does not impact BreakSalv's core gameplay loop, it is left out of this design proposal.

## Administrative & Server Rule Impact (if applicable)

Due to the safety locks on the industrial shredders, the only moderation issue I could see is that an antagonist uses the shredders for DAGD-level destruction by mounting them on a shuttle and trying to bulldoze the station. It does require the EMAG which means it's limited to antagonists only so the risk for self-antag griefing is fairly minor. 

# Technical Considerations

## New features

### Grid dragging

> Before writing this document I had a short sync with Slartibartfast who has experience with grid movement. He said it was likely feasible to implement something like a grappling hook pulling, though it depends on how well Box2D supports continuous forces being applied to a grid. A `DistanceJoint` on the hook may be sufficient however so signs point towards this not being a difficult thing to implement.

Reviewer 4: > though it depends on how well Box2D supports continuous forces being applied to a grid

Pretty sure this is what shuttles do internally

> The main issue likely encountered is that grid mass is currently not very accurate. This would cause problems when determining pull strength and also how the grids should pull towards each other if neither grid on either end of the grappling hook is anchored. If this becomes an issue, grid mass calculations would need a pass to ensure it's more accurate.

Reviewer: ```suggestion
The main issue likely encountered is that grid mass is currently not very accurate. This would cause problems when determining pull strength and also how the grids should pull towards each other if neither grid on either end of the grappling hook is anchored. If this becomes an issue, grid mass calculations would need a pass to ensure they're more accurate.
```

> The main issue likely encountered is that grid mass is currently not very accurate. This would cause problems when determining pull strength and also how the grids should pull towards each other if neither grid on either end of the grappling hook is anchored. If this becomes an issue, grid mass calculations would need a pass to ensure it's more accurate.

Reviewer 4: Our item masses are currently a gigantic meme so we would have to balance everything wholesale.

## Features to be added

BreakSalv is the primary gameplay loop for the _Salvage Specialist_ job belonging to the Salvage subdepartment of the Cargo department. The goal is to produce resources for distribution to the rest of the station, including high volumes of basic materials, rare and unique materials, and equipment that is otherwise difficult to procure. This is achieved by spawning in wrecks of old derelict shuttles, extracting rare materials from them by hand, and then shredding the remainder of the shuttle for basic materials.

The work of a Salvager takes place in the _Salvage Bay_, a location exposed to space on the outside of the station, near the Cargo department. While most Salvage work occurs in the Salvage Bay due to the station-locked machinery there, they should have a small locker room in Cargo for suiting up, with easy access to the rest of the Cargo department to facilitate material transfers.

BreakSalv's gameplay loop consists of three primary parts: retrieval, extraction, and shredding. 

### Retrieval

**Retrieval** is the process of spawning wrecks and transporting them to the station so their resources can be extracted.

> Wrecks are pre-made grids themed around being decommissioned shuttles or destroyed station sections. To access wrecks, Salvage has a unique console in the Salvage Bay called the _Salvage Magnet_. 

Reviewer 5: Yeah, we do have the tech for this already. Right now salvage has two options:
- pre-mapped wrecks which are cool but low in number, work-intensive to create and quickly get extremely repetetive.
- Randomly generated wrecks and asteroids which are extremely bland as they all are just open areas with a few items and mobs thrown on them.

 I think the best approach here is the same as shipbreakers did - premapped ships, but with a randomized interior. And this has a lot of potential for emergent gameplay as the environmental hazards combine in different ways, for example a turret explosion ignites plasma gas, electrovae can be deliberately used to shut down electrical traps and so on.

- **Anchored:** Has a mini-station anchor inside of it that must be manually turned off before the wreck can be moved.
- **Distant:** Spawns at a further distance away from the station than normal. Jetpack recommended.
- **More!** The idea is that these wreck types can be expanded upon for more variety. 

Only one wreck may be active with the magnet at any given time. Once a wreck has been selected, the interface changes to a new view as described in the [Magnet Evaluation](#magnet-evaluation) section.

#### Towing

Once a wreck has been spawned, it will be located somewhere in nearby space in proximity to the Salvage Bay. If the wreck is not within the player's viewing distance, Salvage may use a _Mass Scanner_ mapped in the Bay to locate it. Compared with previous iterations of Salvage, the wrecks are meant to spawn much closer to the station. Regardless, Salvage will want to move the wreck closer to the Salvage Bay, as it will make it easier to extract materials and is necessary to complete the later steps. This is known as _towing_.

> To tow the wreck, Salvage will have access to _grappling hooks_ and _magboots_. While grappling hooks on their own simply drag the user towards the hook, when combined with the grounding force of the magboots, Salvagers can pull grids towards each other (or, in the case of one with a station anchor, only the one without is pulled). Ideally, Salvage will be able to stand on the outer rim of the station and use their grappling hooks to drag the wreck into working distance; once it is, they can proceed to the next step.

Reviewer 6: The salvage "bumbling through space with limited view range" is one of the worse aspects of the job. This does not seem like a good idea at all.

Author: I don't think the scenario which you describe will occur. The document describes that wrecks *should* spawn in view, and if they don't they should be close enough that Salvagers are able to shoot a grappling hook into space to tow it without stepping off the station, reeling it in like a fish.

In the event that the wreck spawns further out, that is cited as an additional challenge parameter.
There is also a feature suggestion to the end of the document describing a "spear gun" which would be a mounted machine on the station to pull the wreck into working distance, though I don't see it a necessitate for the above reasons.

> To tow the wreck, Salvage will have access to _grappling hooks_ and _magboots_. While grappling hooks on their own simply drag the user towards the hook, when combined with the grounding force of the magboots, Salvagers can pull grids towards each other (or, in the case of one with a station anchor, only the one without is pulled). Ideally, Salvage will be able to stand on the outer rim of the station and use their grappling hooks to drag the wreck into working distance; once it is, they can proceed to the next step.

Reviewer 7: Please specify if salvage will have scopes for this, as well as mitigation measures to avoid slamming two grids and causing collision damage.

Wreck resources cannot be used directly as materials; they must be processed first. The _recycler_ is a good option for this, as it is a station-mounted machine, and Salvage can easily have their own. As a rule, wreck resources should _not_ focus on steel or glass materials, since those are primarily obtained through the shredding step.  

- **Two-handed:** Requires two hands to pick up. 
- **Large:** Big enough to require dragging and can't be held or put in containers.
- **Fragile:** Easily broken if thrown around recklessly.
- **Mounted:** Attached to a wall or integrated into some machinery that requires partial deconstruction.
- **Radioactive:** Emits radiation.
- **Electrically Unstable:** If destroyed or processed incorrectly, will emit an EMP pulse.
- **Explosive:** If destroyed or processed incorrectly, will explode.
- **Pulsating:** Occasionally emits a gravity pulse that pushes away other objects around it.
- **More!** The idea is that these wreck resource types can be expanded upon for more variety. 

### Shredding

**Shredding** is the process of finalizing a salvage operation by grinding up the remains of a wreck into base materials.

> Once all valuable wreck resources have been extracted and the Salvagers are finished with the wreck, it is time to dispose of the wreck itself. For this, the Salvage Bay has a large wall of _industrial shredders_, and _separation charges_.

Reviewer 6: This entire concept of shredding entire ships just seems extremely boring. You'd likely be spending quite a lot of time fiddling with the physics and other nonsense just to watch stuff disappear.

Author: From my (admittedly very simple) prototyping, it is gratifying sending chunks of materials into the shredders, hearing the **crunch** and watching materials accumulate. We have that in mini-form with the recycler and I think it is possible to juice up even further.

The main issue I've come across is, as you imply, lining up the grid to fit the shredders. It mostly came down to that the griddrag command doesn't allow for rotation. This was only really a concern when I tried to fit in grids that were just on the brink of being too large.

It's recommended (but not required) that _weld charges_ be implemented as well, which act in opposition to separation charges and enable grid attachment. These would work well as engineering tools and/or as part of the RCD.

#### Magnet Evaluation

The final part of the BreakSalv loop is Salvage returning to the magnet and getting an evaluation of their work.

While a wreck is being worked on, the magnet interface displays a "Magnet Evaluation" view with the following information:

- **Wreck Percentage:** Percentage of how much of the spawned wreck grid has been processed; 0% means all grid tiles remain, 100% means no grid tiles remain.
- **Shredder Efficiency:** Efficiency of connected industrial shredders. This is calculated based on whether the shredders have processed any entities that yield less, such as wreck resources.
- **Wreck Time:** Time since magnet pull started.

> After some time has passed (short enough not to feel restrictive, but long enough that Salvage can't just cycle wrecks), Salvagers may choose to despawn any remaining grids of their wreck using a "Despawn Wreck" button on the magnet. There is, however, an incentive to thoroughly shred the wreck. If the Wreck Percentage is 100%, the previously greyed-out "Claim Processing Reward" button becomes available. Clicking this button rewards Cargo with a monetary boost and announces in the Supply channel that Salvage finished processing the wreck, along with the Shredder Efficiency and Wreck Time. 

Reviewer 6: > If the Wreck Percentage is 100%, the previously greyed-out "Claim Processing Reward" button becomes available.

What is this mechanic meant to encourage? It just feels like you're incentivizing people to spend time doing tedious work (ship shredding) for resources they aren't guaranteed to need (metal & glass).

Author: If we work under the axiom that shredding isn't tedious:

1. Progression through the round. Becomes a way to unlock harder wrecks for a challenge ramp-up. This progression is gated behind Salv actually doing their jobs, whereas before there was nothing stopping Salv from looting the station, building a shuttle and immediately fucking off to vgroid or expeditions which were meant to be late-round content.
2. Affording Salvagers a natural break. It's incorporated after getting feedback from Salvage players that the current systems do not contain any natural pause moments. This is explained in the Game Design Rationale section.

Reviewer 7: I don't shredder efficiency is a good idea.

Material balance will be difficult as efficiency is tied to salvage manpower hours, meaning an overproduction of materials on high-pop servers and vice versa. Wrecks are also already partially randomized in both size and reclaimable materials, which introduces another variation in maintaining material flows.

Encouraging salvagers to work harder will also increase isolation as you're discouraging salvagers from leaving wrecks to keep working.

Reviewer 6: This just isn't a good way to prod salvagers into doing their job. We've *always* had a way to make salvagers "progress": giving them gamer gear (from science or found otherwise). The problem with salvagers fucking off is because:

1. salve fucking sucks lmao nobody would want to play that willingly
2. it's easier to just memorize where gamer gear is mapped than to play the game intentionally

*that* should be fixed, not adding hamfisted mechanics like this.

Once either button has been pressed, the magnet interface returns to the wreck selection view (after a short delay to give a moment to breathe), and Salvage is free to select a new wreck to work on.

### Salvage Progression & Magnet Upgrade

As some wrecks will contain useful tools and equipment to better deal with the three core parts of BreakSalv, there will be a natural progression inherent to Salvage's gameplay. Science may also provide upgraded tools through research and usage of Salvage's materials. Some wrecks will likely be discouraged by being too dangerous or slow to process without these upgrades, lending to a sense of becoming more powerful over the round.

> As a way to add further progression and provide some breathing room for Salvage, BreakSalv also features an upgrade to the salvage magnet's capabilities. This unlocks after Salvage has a certain number of 100% Wreck Percentage completions in the form of a greyed out "Magnet Upgrade" button in the wreck selection view becoming available. Pressing this starts a fairly lenghty progress bar during which no new wrecks can be selected. Salvagers are encouraged to spend this time taking a break on the station and/or visiting the other departments for upgrades.

Reviewer 6: Why are we encouraging ship shredding, again?

Once the progress bar has completed, the Salvagers are rewarded with a new wreck option in the wreck selection view that features a higher-tier wreck. This higher tier is intended to be more challenging by featuring multiple wreck types simultaneously, more resources, and being generally larger.

#### Salvage Specialist equipment

From the start of the round, Salvagers should have access to:

- A salvage hardsuit. Necessary for working in the Salvage Bay.
- Basic tools and toolbelt. For navigating and deconstructing the wreck during the extraction step.
- A survival knife and a portable kinetic accelerator. Used to deal with hostile mobs on wrecks during the extraction step.
- A grappling hook and separation charges. Necessary to complete the retrieval and shredding step.

Upgrades to the Salvager's arsenal could be:

- Improved hardsuit with higher movement speed/defense.
- Welding goggles/gas mask, insulated gloves, and improved welding equipment.
- Jaws of life and power drill.

> - Simple laser weapons.

Reviewer 7: Define 'simple.' How powerful are these and how would they get them if not from science?

## Game Design Rationale

BreakSalv intends to position Salvage as a primary resource generator for the station by leveraging the unique game mechanics opportunities afforded by working outside the station. 

BreakSalv intends to provide the following experience:
- Hard work not for the faint of heart under a sci-fi megacorp industrial aesthetic. Salvagers should feel like they are cool by braving the dangers and efforts inherent to their job.

> - Visible results from their work. Seeing a huge pile of materials come out of the shredders should make the Salvagers feel like they are contributing to the station.

Reviewer 6: > Seeing a huge pile of materials come out of the shredders should make the Salvagers feel like they are contributing to the station.

SS14's departments do not need this many resources. So either this really isn't "huge piles" or it's such huge piles that it's utterly worthless. This already happens with current salvage rounds.

Reviewer 4: Another note is that due to Cargo's current poor design, even if salvage has materials, it's often difficult for departments to get their hands on them in a round for it to actually matter, in my experience.

Author: "Huge" is relative and I feel it is obvious that I am not advocating for gamebreaking levels of overabundance. The return the shredders give can be tweaked and the goal when making the wrecks is to have better understanding of just how much resources shredding one is worth.

Writing "Seeing piles of balanced amounts of materials as to not trivialize future resource gathering but not so small as to make the activity useless should make Salvagers feel line they are contributing to the station" isn't gonna read as punchy.

Author: I agree with Roomba's comment, but also see it as OoS for this document.

Reviewer 7: See earlier comment regarding shredding efficiency.

> - Completionism and optimization. Salvagers will want to hit 100% on the Wreck Percentage with high Shredder Efficiency and low Wreck Time, but may situationally forgo it for a faster, lower reward. Good Salvagers can brag about how fast they completed a wreck.

Reviewer 6: This is explicitly and commonly known to be a ***bad thing*** in game design. Going for 100% completion is *not generally* considered fun, and often just results in extreme amounts of tedium with little gain.

Reviewer 7: See earlier comment regarding shredding efficiency. Salvagers will already progress via science and magnet upgrades.

- Moments of rest. Since all Salvage work requires constant manual effort, salvagers shouldn't be pressured to work continuously. Not having a time limit and forcing minor delays between wrecks (and a longer one for the Magnet Upgrade) are meant to encourage this.

There are a few things that BreakSalv is intentionally trying to avoid:

> - Isolationism. While Salvage operates on the outside of the station, since the focus is on magnet wrecks and the shredders, Salvagers do not have much reason to leave the station proper. At worst, a Salvage team may occasionally need to leave the station grid to retrieve a wreck that has spawned further out, but this is only intended to be a temporary excursion lasting for the duration to get out there and tow it back. Salvage should be expected to be found in the Salvage Bay at a level similar to Botany and Xenoarch.

Reviewer 7: Isolation is also bad for antagonist balance, whether your target is a salvager or you are a salvage traitor.

## Modified features

### Station map considerations

Station maps would need to accommodate the new "shredder wall" concept. After briefly testing, it seems that a width of 9 tiles (the width of a singularity containment field) is around the ideal size to deal with pulling in wrecks for shredding. Most stations seem to have room to fit them without issue (except possibly Reach, though even there, there is room for a smaller version).

Still, all maps would need to implement a new Salvage Bay to be able to support BreakSalv. 

### Wreck map considerations

Many, if not all, of the existing wreck maps would likely need to be redesigned to comply with the wreck type system and provide an engaging wreck for Salvage to process in the manner that BreakSalv requires. The magnet must also be able to properly track how much of a spawned grid remains, which includes whether the grid is split off into multiple parts. This should be doable via a simple tracking component, hopefully.

### Salvage Magnet

> The salvage magnet would require a partial re-design with its UI to support the new functionality. Most of what exists could probably be repurposed for BreakSalv however.

Reviewer 7: The magnet should work only on the station grid.

The magnet also needs a green and red light to indicates if there's enough space for the wreck to be pulled.
